WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.kt

```diff
@@ -0,0 +1,130 @@
+package org.wordpress.android.ui.posts
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.appcompat.view.ContextThemeWrapper
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.databinding.AddCategoryBinding
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.TermModel
+import org.wordpress.android.fluxc.store.TaxonomyStore
+import org.wordpress.android.models.CategoryNode
+import org.wordpress.android.util.ToastUtils
+import javax.inject.Inject
+
+class AddCategoryFragment : AppCompatDialogFragment() {
+    private var site: SiteModel? = null
+    private var binding: AddCategoryBinding? = null
+
+    @set:Inject
+    var mTaxonomyStore: TaxonomyStore? = null
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        (requireActivity().application as WordPress).component().inject(this)
+        initSite(savedInstanceState)
+        val builder =
+            MaterialAlertDialogBuilder(ContextThemeWrapper(activity, R.style.PostSettingsTheme))
+        binding = AddCategoryBinding.inflate(layoutInflater, null, false)
+        loadCategories()
+        builder.setView(binding!!.root)
+            .setPositiveButton(android.R.string.ok, null)
+            .setNegativeButton(android.R.string.cancel, null)
+
+        return builder.create()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        val dialog = requireDialog() as AlertDialog
+        dialog.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener {
+            if (addCategory()) {
+                dismiss()
+            }
+        }
+    }
+
+    @Suppress("Deprecation")
```

Comment:
We shouldn't add any new `@Suppress` when migrating to Kotlin. You can use [Bundle.getSerializableCompat()](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/extensions/CompatExtensions.kt#L76) for `getSerializable()`. Check usages of it in other Kotlin classes.

Reply:
Fixed in the latest commit 👍